src/menu/file_select.c

```diff
@@ -1887,65 +2011,78 @@ void print_score_file_star_score(s8 fileIndex, s16 courseIndex, s16 x, s16 y) {
     }
     // Add 1 star character for every star collected
     for (i = 0; i < starCount; i++) {
-        starScoreText[i] = DIALOG_CHAR_STAR_FILLED;
+        entries[i] = "★";
     }
-    // Terminating byte
-    starScoreText[i] = DIALOG_CHAR_TERMINATOR;
+    for (i = starCount; i < 6; i++) {
+        entries[i] = "";
+    }
+    sprintf(starScoreText, "%s%s%s%s%s%s", entries[0], entries[1], entries[2], entries[3], entries[4], entries[5]);
     print_menu_generic_string(x, y, starScoreText);
 }
 
-#define MARIO_X         25
-#define FILE_LETTER_X   95
-#define LEVEL_NUM_PAD    3
-#define SECRET_STARS_PAD 6
-#define LEVEL_NAME_X    23
-#define STAR_SCORE_X   171
-#define MYSCORE_X      238
-#define HISCORE_X      231
+langarray_t textScoreMenuMarioX = DEFINE_LANGUAGE_ARRAY(
+    "MARIO %s",
+    "MARIO %s",
+    "MARIO %s",
+    "マリオ %s");
+
+langarray_t textHiScore = DEFINE_LANGUAGE_ARRAY(
+    "HI SCORE",
+    "MEILLEUR SCORE",
+    "BESTLEISTUNG",
+    "ハイスコア");
+
+extern langarray_t textMyScore;
 
 /**
  * Prints save file score strings that shows when a save file is chosen inside the score menu.
  */
 void print_save_file_scores(s8 fileIndex) {
     u32 i;
-    unsigned char textMario[] = { TEXT_MARIO };
-    unsigned char textHiScore[] = { TEXT_HI_SCORE };
-    unsigned char textMyScore[] = { TEXT_MY_SCORE };
-    unsigned char textFileLetter[] = { TEXT_ZERO };
-    void **levelNameTable = segmented_to_virtual(languageTable[gInGameLanguage][1]);
+    char str[20];
+    char fileLetter[4];
```

Comment:
No need for this to be an array, `%c` should work with `sprintf` for single characters.

Reply:
ah yeah that makes sense